src/core/server/saved_objects/service/lib/delete_index_templates.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ElasticsearchClient } from '../../../elasticsearch';
+import type { Logger } from '../../../logging';
+
+interface DeleteIndexTemplatesOptions {
+  client: ElasticsearchClient;
+  log: Logger;
+}
+
+/**
+ * Deletes old index templates that were used in 6.x as those will no longer be supported in ES 8.
+ */
+export const deleteIndexTemplates = async ({ client, log }: DeleteIndexTemplatesOptions) => {
+  const { body: templates } = await client.cat.templates({
+    format: 'json',
+    name: 'kibana_index_template*',
+  });
+
+  if (!templates.length) {
+    return;
+  }
+  const templateNames = templates
+    .filter((template) => template.name)
+    .map((template) => template.name!);
+
+  log.debug(`Deleting index templates: ${templateNames.join(', ')}`);
+
+  return await Promise.all(
+    templateNames.map((templateName) => {
+      return client.indices.deleteTemplate({
+        name: templateName,
+      });
+    })
+  );
```

Comment:
This is currently throwing if a call to `client.indices.deleteTemplate` fails for any reason. I wonder if we should be more conservative and ignore the potential errors?

Reply:
yeah especially when two Kibana's both see the template on startup, because only one will succeed in deleting. So I think it's fine to wrap the call in a try catch and ignore any errors.

Reply:
done in https://github.com/elastic/kibana/pull/119150/commits/273ccdf82f716176a36a37a3a076a2d0bb67d8c7